localstack/services/apigateway/integration.py

```diff
@@ -254,47 +335,35 @@ def invoke(self, invocation_context: ApiInvocationContext):
             invocation_context.context["authorizer"] = invocation_context.auth_context
 
         request_templates = RequestTemplates()
-        payload = request_templates.render(invocation_context)
-
-        invocation_result = lambda_api.run_lambda(
-            func_arn=func_arn,
-            event=payload,
+        event = request_templates.render(invocation_context) or b""
+        asynchronous = headers.get("X-Amz-Invocation-Type", "").strip("'") == "Event"
+        result = call_lambda(
+            function_arn=func_arn, event=to_bytes(event), asynchronous=asynchronous
         )
-        result = invocation_result.result
-
-        if isinstance(result, FlaskResponse):
-            response = flask_to_requests_response(result)
-        elif isinstance(result, Response):
-            response = result
-        else:
-            response = LambdaResponse()
 
-            is_async = headers.get("X-Amz-Invocation-Type", "").strip("'") == "Event"
+        response = LambdaResponse()
 
-            if is_async:
-                response._content = ""
-                response.status_code = 200
-            else:
-                # depending on the lambda executor sometimes it returns a string and sometimes a dict
-                match result:
-                    case dict():
+        if asynchronous:
+            response._content = ""
+            response.status_code = 200
+        else:
+            # depending on the lambda executor sometimes it returns a string and sometimes a dict
+            match result:
+                case str():
+                    # try to parse the result as json, if it succeeds we assume it's a valid
+                    # json string, and we don't do anything.
+                    if isinstance(json.loads(result or "{}"), dict):
                         parsed_result = result
-                    case str():
```

Comment:
Sometimes, particularly dependent on the executor runtime, the response payload would be a dictionary or a string. For the string cases it could come in both ways:

```
'{"msg":"hello world"}'
```
or, notice the wrapper string around the json string. 
```
"'{"msg":"hello world"}'"
```
Thus, I think is secure to remove this code only if this behaviour is not dependent on the lambda executor.

Reply:
Good point, when going over this I thought it should not happen if its properly going through the serialization, not too sure though to be honest. With the new lambda provider we can definitely expect correct responses here, but I will readd this logic for now!

Reply:
After looking into it, I did not delete the respective code parts, its just weirdly formatted for github (due to changed indentation I assume). The following code block (which is still in there) should cover both cases:
```
                    # try to parse the result as json, if it succeeds we assume it's a valid
                    # json string, and we don't do anything.
                    if isinstance(json.loads(result or "{}"), dict):
                        parsed_result = result
                    else:
                        # the docker executor returns a string wrapping a json string,
                        # so we need to remove the outer string
                        parsed_result = json.loads(result or "{}")
```
Only the "its a dict before" case is now deleted, as the input is always a string now.
With the Lambda rework, we definitely need to address this behavior, though.